pkg/didcomm/protocol/didexchange/states.go

```diff
@@ -441,6 +453,89 @@ func (ctx *context) handleInboundRequest(request *Request, options *options,
 	}, connRec, nil
 }
 
+func (ctx *context) prepareResponse(request *Request, responseDidDoc *did.Doc,
+	responseConnection *Connection) (*Response, error) {
+	// prepare the response
+	response := &Response{
+		Type: ResponseMsgType,
+		ID:   uuid.New().String(),
+		Thread: &decorator.Thread{
+			ID: request.ID,
+		},
+	}
+
+	if request.Thread != nil {
+		response.Thread.PID = request.Thread.PID
+	}
+
+	if !ctx.doInterop {
```

Comment:
I suggest renaming the flag to `doACAPyInterop` to make the intention clearer.

Reply:
Done